### PR TITLE
[Networking] Tiny refactoring of p2pNode

### PR DESCRIPTION
**Pull request by nozim**
Grouping Libp2pNode operations into smaller interfaces

### CODE REVIEW

**Comment by nozim** [comment]
Re: https://github.com/onflow/flow-go/pull/4630

File: network/p2p/libp2pNode.go
Comment (yhassanzadeh13) [inline network/p2p/libp2pNode.go:41]: `WithDefaultUnicastProtocol` does not belong to the `PeerManagement` domain, it is more relevant to `CreateStream` method.

File: network/p2p/libp2pNode.go
Comment (yhassanzadeh13) [inline network/p2p/libp2pNode.go:93]: `flow` is recommended to be written as a noun, i.e., `Flow`.

File: network/p2p/libp2pNode.go
Comment (yhassanzadeh13) [inline network/p2p/libp2pNode.go:99]:
```suggestion

```

File: network/p2p/libp2pNode.go
Comment (yhassanzadeh13) [inline network/p2p/libp2pNode.go:135]:
```suggestion
	UnicastManagement
```

File: network/p2p/libp2pNode.go
Comment (yhassanzadeh13) [inline network/p2p/libp2pNode.go:23]:
```suggestion
type LibP2PNode interface {
```
Otherwise will be confused with other definitions of node in our codebase.

Comment (nozim) [reply network/p2p/libp2pNode.go]: https://github.com/onflow/flow-go/pull/4630/commits/455147a08f62bace4ff3fb18c2cc6ae018d3ca68

File: network/p2p/libp2pNode.go
Comment (nozim) [reply network/p2p/libp2pNode.go]: https://github.com/onflow/flow-go/pull/4630/commits/455147a08f62bace4ff3fb18c2cc6ae018d3ca68

File: network/p2p/libp2pNode.go
Comment (nozim) [reply network/p2p/libp2pNode.go]: https://github.com/onflow/flow-go/pull/4630/commits/455147a08f62bace4ff3fb18c2cc6ae018d3ca68

File: network/p2p/libp2pNode.go
Comment (nozim) [reply network/p2p/libp2pNode.go]: https://github.com/onflow/flow-go/pull/4630/commits/455147a08f62bace4ff3fb18c2cc6ae018d3ca68

File: network/p2p/libp2pNode.go
Comment (nozim) [reply network/p2p/libp2pNode.go]: https://github.com/onflow/flow-go/pull/4630/commits/455147a08f62bace4ff3fb18c2cc6ae018d3ca68

File: network/p2p/libp2pNode.go
Comment (nozim) [reply network/p2p/libp2pNode.go]: https://github.com/onflow/flow-go/pull/4630/commits/455147a08f62bace4ff3fb18c2cc6ae018d3ca68

File: network/p2p/libp2pNode.go
Comment (nozim) [reply network/p2p/libp2pNode.go:33]: https://github.com/onflow/flow-go/pull/4630/commits/455147a08f62bace4ff3fb18c2cc6ae018d3ca68

**Comment by nozim** [comment]
@yhassanzadeh13 addressed all the feedback. Kindly re-review.

File: network/p2p/libp2pNode.go
Comment (yhassanzadeh13) [inline network/p2p/libp2pNode.go:22]:
```suggestion
type CoreP2P interface {
```

**Review by yhassanzadeh13 — APPROVED** [review]
Looks good, thanks for applying the comments, please also apply one minor comment below before merging.

File: network/p2p/libp2pNode.go
Comment (nozim) [reply network/p2p/libp2pNode.go]: https://github.com/onflow/flow-go/pull/4630/commits/7c719df145fea69a73e9baf6ddaa738433f72b50

**Comment by nozim** [comment]
@yhassanzadeh13 kindly re-review and run CI.

**Comment by yhassanzadeh13** [comment]
> @yhassanzadeh13 kindly re-review and run CI.

@nozim we need another review before merge (2 approvals required), once @kc1116 approved your PR, we can run the CI. Thanks!

**Comment by nozim** [comment]
@yhassanzadeh13 @kc1116 kindly re-review. Fixed some compilation errors.

File: network/p2p/p2pnode/libp2pNode.go
Comment (kc1116) [inline network/p2p/p2pnode/libp2pNode.go:93]: Why was this removed?

Comment (nozim) [reply network/p2p/p2pnode/libp2pNode.go:93]: its already declared here https://github.com/onflow/flow-go/blob/9c5b9f154064b543b01031a82afbd1c7846ead5b/network/p2p/p2pnode/libp2pNode.go#L51

Comment (yhassanzadeh13) [reply network/p2p/p2pnode/libp2pNode.go:93]: @nozim, kindly allow the reviewer, in this case @kc1116, to resolve the conversation. When the PR author unilaterally resolves a conversation, it complicates the tracking of the PR's progress.

**Review by yhassanzadeh13 — APPROVED** [review]

**Review by kc1116 — APPROVED** [review]

File: network/p2p/p2pnode/libp2pNode.go
Comment (nozim) [reply network/p2p/p2pnode/libp2pNode.go:93]: my apologies @yhassanzadeh13 and @kc1116. Will let the reviewer resolve future feedback.

**Pull request merged**
